src/build/_util.py

```diff
@@ -13,8 +14,17 @@
 )
 
 
-def check_dependency(
-    req_string: str, ancestral_req_strings: tuple[str, ...] = (), parent_extras: Set[str] = frozenset()
+# https://www.python.org/dev/peps/pep-0503/#normalized-names
+def _normalize(name: str) -> str:
+    return re.sub(r'[-_.]+', '-', name).lower()
+
+
+def check_dependency(  # noqa: C901
+    req_string: str,
+    ancestral_req_strings: tuple[str, ...] = (),
+    parent_extras: Set[str] = frozenset(),
+    project_name: str | None = None,
```

Comment:
You don't need a separate `project_name` - you just need to pass the project name as an `ancestral_req_strings` 1-tuple.

Reply:
This approach seems like it might be problematic due to the `CircularBuildDependencyError` exception class having special handling for `self.backend` which is used for tracing circular build dependencies via the `get_requires_for_build_{distribution}` hooks.

Reply:
You'll have to explain why it's problematic.

Reply:
Ignore this actually.  Circular runtime deps are no factor - we can't repurpose `ancestral_req_strings`.